dbt/adapters/redshift/connections.py

```diff
@@ -208,14 +210,22 @@ def connect():
                 )
 
             def connect():
+                if access_key_id := self.credentials.access_key_id:
```

Comment:
Allow for both the file approach, and the direct approach.

Reply:
we should make sure to document this (even if it's intuitive) that providing an access_key_id overrides your local AWS cli setup

Reply:
also we should explicitly check/raise an exception for providing both a `secret_access_key` and a `access_key_id` (i.e. if you provide one then you need to provide both)

Reply:
> we should make sure to document this (even if it's intuitive) that providing an access_key_id overrides your local AWS cli setup

Good point, given we're effectively adding an auth method here, we need docs either way. But we need to explicitly call out the order in which `dbt-redshift` will look for creds.

> also we should explicitly check/raise an exception for providing both a secret_access_key and a access_key_id (i.e. if you provide one then you need to provide both)

I thought about doing this and didn't because python is a leap without looking language. However, I didn't think about what this looks like for the user. I'll add that in.

Reply:
Along those lines, should we also add checks for providing both a profile and the explicit creds? Or simply document the order?

Reply:
This was added